internal/azureclient/client.go

```diff
@@ -306,6 +313,45 @@ func (c *AzureClient) GetLoadBalancer(ctx context.Context, subscriptionID, resou
 	return lb, nil
 }
 
+// GetPrivateEndpoint retrieves information about the specified private endpoint.
+func (c *AzureClient) GetPrivateEndpoint(ctx context.Context, subscriptionID, resourceGroup, peName string) (*armnetwork.PrivateEndpoint, error) {
+	// Create cache key
+	cacheKey := fmt.Sprintf("resource:privateendpoint:%s:%s:%s", subscriptionID, resourceGroup, peName)
+
+	// Check cache first
+	if cached, found := c.cache.Get(cacheKey); found {
+		if pe, ok := cached.(*armnetwork.PrivateEndpoint); ok {
+			return pe, nil
+		}
+	}
+
+	clients, err := c.GetOrCreateClientsForSubscription(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.PrivateEndpointsClient.Get(ctx, resourceGroup, peName, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get private endpoint: %v", err)
+	}
+
+	pe := &resp.PrivateEndpoint
+	// Store in cache
+	c.cache.Set(cacheKey, pe)
+
+	return pe, nil
+}
+
+// GetPrivateEndpointByID retrieves information about the specified private endpoint using its resource ID.
+func (c *AzureClient) GetPrivateEndpointByID(ctx context.Context, privateEndpointID string) (*armnetwork.PrivateEndpoint, error) {
+	parsedID, err := arm.ParseResourceID(privateEndpointID)
```

Comment:
No validation that the parsed resource ID is actually for a private endpoint.

Reply:
fixed.